src/rgw/rgw_sal_motr.cc

```diff
@@ -2326,6 +2326,32 @@ int MotrObject::delete_obj_aio(const DoutPrefixProvider* dpp, RGWObjState* astat
   return 0;
 }
 
+int MotrGetDataCB_Wrapper::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len) {
+  uint64_t chunk_size = s->cct->_conf->rgw_max_chunk_size;
+  uint64_t chunk_len, buf_len=0;
+  bufferptr bp(bl.c_str() + bl_ofs, bl_len);
+  bl_tmp.push_back(bp);
+  buf_len = bl_tmp.length();
+  bl_ofs=0;
+
+  do {
+    chunk_len = chunk_size > buf_len ? buf_len : chunk_size;
+    cb->handle_data(bl_tmp, bl_ofs, chunk_len);
+    buf_len -= chunk_len;
+    bl_ofs += chunk_len;
+    if (buf_len<chunk_size && !last) {
```

Comment:
Hi, will this ever occur? where buf_len is less than chunk_size and it is not the last chunk?

Reply:
This can happen when bl_len is not divisible by chunk_size and there are more data to be sent. 
ex: if object size is 30Mb, bs=9Mb, chunk_size=4Mb, in this case it sends 4Mb chucks in first two iterations and for the third iteration 1Mb data is left. So instead of doing a network IO for only 1Mb data we can store it in bl_tmp and it will send it in the next handle_data() call from read_mobj() along with next 9Mb (bs) data. This will optimize network IO. And only for the last remaining block of data it will do network IO even if the remaining data is less than chunk_size.

Reply:
Ohh okay great. 👍🏼